src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -284,18 +293,28 @@ public boolean allocateUnassigned(RoutingAllocation allocation) {
             }
 
             if (!canBeAllocatedToAtLeastOneNode) {
+                // still fetching data, remove from the unassigned, and try the next
+                unassignedIterator.remove();
+                routingNodes.ignoredUnassigned().add(shard);
```

Comment:
I will do this in another change, I want to change the allocator as little as possible with this change to easily backport